vcd/resource_vcd_external_network.go

```diff
@@ -0,0 +1,381 @@
+	if err != nil {
+		return err
+	}
+
+	task, err := govcd.CreateExternalNetwork(vcdClient.VCDClient, params)
+	if err != nil {
+		log.Printf("[DEBUG] Error creating external network: %#v", err)
+		return fmt.Errorf("error creating external network: %#v", err)
+	}
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		log.Printf("[DEBUG] Error waiting for external network to finish: %#v", err)
+		return fmt.Errorf("error waiting for external network to finish: %#v", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+	log.Printf("[TRACE] external network created: %#v", task)
+	return resourceVcdExternalNetworkRead(d, meta)
+}
+
+// Fetches information about an existing external network for a data definition
+func resourceVcdExternalNetworkRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] external network read initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	externalNetwork, err := govcd.GetExternalNetwork(vcdClient.VCDClient, d.Id())
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("error fetching external network details %#v", err)
+	}
+
+	log.Printf("[TRACE] external network read completed: %#v", externalNetwork.ExternalNetwork)
+	return nil
+}
+
+// Deletes a external network, optionally removing all objects in it as well
+func resourceVcdExternalNetworkDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] external network delete started")
+
+	vcdClient := meta.(*VCDClient)
+
+	externalNetwork, err := govcd.GetExternalNetwork(vcdClient.VCDClient, d.Id())
+	if err != nil {
+		log.Printf("[DEBUG] Error fetching external network details %#v", err)
+		return fmt.Errorf("error fetching external network details %#v", err)
+	}
+
+	err = externalNetwork.DeleteWait()
+	if err != nil {
+		log.Printf("[DEBUG] Error removing external network %#v", err)
+		return fmt.Errorf("error removing external network %#v", err)
+	}
+
+	log.Printf("[TRACE] external network delete completed: %#v", externalNetwork)
+	return nil
+}
+
+// helper for transforming the resource input into the ExternalNetwork structure
+// any cast operations or default values should be done here so that the create method is simple
+func getExternalNetworkInput(d *schema.ResourceData, vcdClient *VCDClient) (*types.ExternalNetwork, error) {
+	params := &types.ExternalNetwork{
+		Name:        d.Get("name").(string),
+		Xmlns:       types.XMLNamespaceExtension,
+		XmlnsVCloud: types.XMLNamespaceVCloud,
+		Configuration: &types.NetworkConfiguration{
+			Xmlns:                          types.XMLNamespaceVCloud,
+			FenceMode:                      d.Get("fence_mode").(string),
+			RetainNetInfoAcrossDeployments: d.Get("retain_net_info_across_deployments").(bool),
+		},
+	}
+
+	ipScopesConfigurations := d.Get("ip_scope").([]interface{})
+	// This is a limitation from the vcloud package
+	/*	if len(ipScopes) != 1 {
+		return &types.ExternalNetwork{}, fmt.Errorf("only one ip_scope is allowed")
+	}*/
```

Comment:
Do we need this commented code? Does it still says the truth? In the GUI you can add many "network settings" as they call it.

Reply:
Removed commented out code as limitation was changed to support multi scopes